### PR TITLE
Static dosmaps

### PR DESCRIPTION
**Pull request by MaxHellstrom**
Solves #15

### CODE REVIEW

File: src/pyskindose/analyze_data.py [hunk 1]
```diff
@@ -56,9 +56,11 @@ def analyze_data(
     if not plot_dose_map:
         return output
 
-    create_dose_map_plot(
-        patient=patient, settings=settings,
-        dose_map=output[c.OUTPUT_KEY_DOSE_MAP]
-    )
+    if settings.mode == c.MODE_CALCULATE_DOSE:
```
Comment (BwKodex) [inline src/pyskindose/analyze_data.py:59]: What should happen here if `settings.mode != c.MODE_CALCULATE_DOSE`?

If this is the only setting allowed here you should raise an exception if `settings.mode` has another value

File: src/pyskindose/constants.py [hunk 2]
```diff
@@ -146,12 +147,16 @@
 PLOT_ASPECTMODE_PLOT_DOSEMAP = 'data'
 
 PLOT_ASPECTMODE_PLOT_PROCEDURE = 'cube'
+PLOT_ORDER_STATIC = ['right', 'back', 'left', 'front']
 
 PLOT_PROCEDURE_AXIS_RANGE_X = [-300, 300]
 PLOT_PROCEDURE_AXIS_RANGE_Y = [-300, 300]
 PLOT_PROCEDURE_AXIS_RANGE_Z = [-300, 300]
-
+PLOT_GROUND_SHIFT_X_STATIC = 50
```
Comment (BwKodex) [inline src/pyskindose/constants.py:155]: Perhaps we could add comments specifying where these "magic" numbers come from? Might be helpful for later

File: src/pyskindose/plotting/create_dose_map_plot.py [hunk 3]
```diff
@@ -1,21 +1,256 @@
...
+            hoverinfo='text',
+            text=hover_text, name="Human",
+            colorbar=dict(tickfont=dict(color=COLOR_PLOT_TEXT),
+                          title="Skin dose [mGy]",
+                          titlefont=dict(
+                            family=PLOT_FONT_FAMILY,
+                            color=COLOR_PLOT_TEXT)))]
+
+    # Layout settings
+    layout = go.Layout(
```
Comment (BwKodex) [inline src/pyskindose/plotting/create_dose_map_plot.py:105]: Could we extract the layout creation to a separate function to make this one more lean?

File: src/pyskindose/plotting/create_dose_map_plot.py [hunk 4]
```diff
@@ -1,21 +1,256 @@
...
+                )
+        )
+
+    # create figure
+    fig = go.Figure(data=phantom_mesh, layout=layout)
+
+    if settings.plot.interactivity:
+        fig.show()
+
+    else:
```
Comment (BwKodex) [inline src/pyskindose/plotting/create_dose_map_plot.py:161]: If you add a `return` after `fig.show()` above you can skip this else and reduce the nesting

File: src/pyskindose/plotting/create_dose_map_plot.py [hunk 5]
```diff
@@ -1,21 +1,256 @@
...
+            PLOT_EYE_RIGHT, PLOT_EYE_BACK, PLOT_EYE_LEFT, PLOT_EYE_FRONT]
+
+        names = PLOT_ORDER_STATIC
+        file_type_static = PLOT_FILE_TYPE_STATIC
+
+        for i in pbar(range(len(eyes)), desc=f'saving static dosemaps: '):
+            fig['layout']['scene']['camera'] = eyes[i]
+            fig.write_image(f"{names[i]}.png")
+
+        if settings.plot.notebook_mode:
```
Comment (BwKodex) [inline src/pyskindose/plotting/create_dose_map_plot.py:178]: Perhaps extract the plotting for `notebook_mode` done here into a separate function?

File: src/pyskindose/plotting/create_dose_map_plot.py [hunk 6]
```diff
@@ -1,21 +1,256 @@
...
+
+            fig.update_layout(
+                width=total_width,
+                height=total_height,
+                margin={"l": 0, "r": 0, "t": 0, "b": 0},
+            )
+
+            fig.show(config={'doubleClick': 'reset'})
+
+        if not settings.plot.notebook_mode:
```
Comment (BwKodex) [inline src/pyskindose/plotting/create_dose_map_plot.py:252]: Same here as above. If you add a return after `fig.show(config={'doubleClick': 'reset'}) above you can skip this if statement.

File: tests/manual_tests/notebook_tests/notebook_test_calculate_dose_interactive.ipynb [hunk 7]
```diff
@@ -84,7 +84,7 @@
    "name": "python",
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
-   "version": "3.7.4"
+   "version": "3.9.0"
```
Comment (BwKodex) [inline tests/manual_tests/notebook_tests/notebook_test_calculate_dose_interactive.ipynb:87]: Nice with the Python upgrade!

File: src/pyskindose/settings_pyskindose.py [hunk 8]
```diff
@@ -71,12 +71,24 @@ def __init__(self, settings: Union[str, dict]):
...
         self.mode = tmp[KEY_PARAM_MODE]
+        self.k_tab_val = tmp[KEY_PARAM_K_TAB_VAL]
         self.rdsr_filename = tmp[KEY_PARAM_RDSR_FILENAME]
         self.estimate_k_tab = tmp[KEY_PARAM_ESTIMATE_K_TAB]
-        self.k_tab_val = tmp[KEY_PARAM_K_TAB_VAL]
+
         self.phantom = PhantomSettings(ptm_dim=tmp['phantom'])
         self.plot = Plotsettings(plt_dict=tmp['plot'])
 
+    def print_parameters(self):
```
Comment (BwKodex) [inline src/pyskindose/settings_pyskindose.py:81]: Could you add a unit test to test this print functionality?

**Review by BwKodex — CHANGES_REQUESTED** [review]

File: src/pyskindose/analyze_data.py (hunk 1 above)
Comment (MaxHellstrom) [reply src/pyskindose/analyze_data.py]: Resolved in fa2ee270b67e37a926235a83d687cf7fc0897188

File: src/pyskindose/constants.py (hunk 2 above)
Comment (MaxHellstrom) [reply src/pyskindose/constants.py:155]: Resolved in 41fa517539332949b571d0a07ece664a7ffbc0b7

File: src/pyskindose/plotting/create_dose_map_plot.py (hunk 4 above)
Comment (MaxHellstrom) [reply src/pyskindose/plotting/create_dose_map_plot.py:179]: Resolved in c27f2095733acffaee2458dee282612628288355

File: src/pyskindose/plotting/create_dose_map_plot.py (hunk 6 above)
Comment (MaxHellstrom) [reply src/pyskindose/plotting/create_dose_map_plot.py:269]: resolved in c97f5df038baa319b97d4b60fb160caee6e99777

Comment (MaxHellstrom) [reply src/pyskindose/plotting/create_dose_map_plot.py]: resolved in c97f5df038baa319b97d4b60fb160caee6e99777

File: src/pyskindose/settings_pyskindose.py (hunk 8 above)
Comment (MaxHellstrom) [reply src/pyskindose/settings_pyskindose.py]: Sure, added in 6dcbf4c7e495c6d1a5bc9ea9201498b92dd5ba21

File: src/pyskindose/plotting/create_dose_map_plot.py (hunk 6 above)
Comment (MaxHellstrom) [reply src/pyskindose/plotting/create_dose_map_plot.py]: resolved (again 👯 ) in c9a4ae2f60becbeb6ca4bce2d520fa03710f50ca

File: src/pyskindose/plotting/create_dose_map_plot.py (hunk 5 above)
Comment (MaxHellstrom) [reply src/pyskindose/plotting/create_dose_map_plot.py]: Yes! dc2f659f56a27695e3b95966e686a332157ec0af

File: src/pyskindose/plotting/create_dose_map_plot.py (hunk 3 above)
Comment (MaxHellstrom) [reply src/pyskindose/plotting/create_dose_map_plot.py]: yes, resolved in a1dfe919b9f1549a96b9951c1c250b86722f2613

**Review by BwKodex — APPROVED** [review]

File: tests/unittests/test_settings.py [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+
+P = Path(__file__).parent.parent
+sys.path.insert(1, str(P.absolute()))
+
+from pyskindose.settings_pyskindose import PyskindoseSettings
+from manual_tests.base_dev_settings import DEVELOPMENT_PARAMETERS
+
+settings = PyskindoseSettings(settings=DEVELOPMENT_PARAMETERS)
+
+
```
[the remark this answers is not in the document]
Comment (MaxHellstrom) [reply tests/unittests/test_settings.py:12]: yes, this was added in 6dcbf4c7e495c6d1a5bc9ea9201498b92dd5ba21

**Review by BwKodex — APPROVED** [review]

**Pull request merged**